pylint/lint.py

```diff
@@ -1054,31 +1083,51 @@ def _do_check(self, files_or_modules):
             if interfaces.implements(checker, interfaces.IAstroidChecker):
                 walker.add_checker(checker)
         # build ast and check modules or packages
-        for descr in self.expand_files(files_or_modules):
-            modname, filepath, is_arg = descr["name"], descr["path"], descr["isarg"]
-            if not self.should_analyze_file(modname, filepath, is_argument=is_arg):
-                continue
+        if self.config.from_stdin:
+            assert len(files_or_modules) == 1
```

Comment:
I'm not sure I understand why we need a file here? Can't we output the report to the stdout? If we need a filename for reporting purposes, wouldn't be enough to provide a fake one, e.g. `<string>` or something of sorts?

Reply:
I've seen that providing a fake filename is also supported by other linters
See e.g. https://github.com/jshint/jshint/pull/1503 and https://eslint.org/docs/user-guide/command-line-interface

Reply:
I think this matters in particular for (relative) imports.

Reply:
You are right, handling of relative imports needs to be taken into account. I'll write a testcase for that and then we can discuss how to fix the issue.

Reply:
here is the testcase https://github.com/thisch/pylint/commit/5583543d9c9e780ea84dde0697e38b7ccafea24a

the results output by the `--from-stdin` run differ from the normal (non-stdin) run.

Can anybody explain why the relative imports are not working in the `--from-stdin` case?

Reply:
I've just fixed a bug in the code that determines the module name of the file passed as a positional argument to pylint (after --from-stdin).

To determine the correct modulename I copied the code from pylint.utils.expand_modules. I hope that I didn't miss anything.

The ``--from-stdin`` feature should now work as expected.